### PR TITLE
fix: androidTest lint

### LINKED ISSUE #10322: [Bug] Lint is not run on `androidTest`
###### Reproduction Steps

1. On the latest master, or a branch off the latest master (I am using [0184435](https://github.com/viciousAegis/Anki-Android/commit/01844357aa09a4c3778f30c4d7ea7798bba71f1d) as my latest pull)
2. run a lint check 

Not sure if this is a problem with my local repo, or the remote one; so posting here

###### Expected Result
Check gets completed with no building errors, and all the lint issues are displayed

###### Actual Result
While lint issues are displayed (not sure if all of them are), the following error arises in the Terminal for me:
````
> Task :AnkiDroid:lintAmazonDebug FAILED
/*
Lint issues listed
*/
27 errors, 0 warnings

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':AnkiDroid:lintAmazonDebug'.
> Lint found errors in the project; aborting build.
  
  Fix the issues identified by lint, or create a baseline to see only new errors:
  ```
  android {
      lint {
          baseline = file("lint-baseline.xml")
      }
  }
  ```
  
  For more details, see https://developer.android.com/studio/write/lint#snapshot

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1s
60 actionable tasks: 3 executed, 1 from cache, 56 up-to-date
````
On checking the `Inspect Code` Output for lint errors:
image

###### Debug info
```
MacOs 12.0.1
Android Studio Bumblebee | 2021.1.1
Build #AI-211.7628.21.2111.8092744, built on January 19, 2022
Runtime version: 11.0.11+0-b60-7772763 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.0.1
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 8
Registry: external.system.auto.import.disabled=true
Non-Bundled Plugins: cn.haojiyou.CodeGlance3 (2.1.0), org.jetbrains.kotlin (211-1.6.10-release-923-AS7442.40), org.intellij.plugins.markdown (211.7142.37)
```
###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
Fixes lint errors, then fixes `lint.yml` to run `androidTest`

## Issue
Fixes #10322

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ACRATest.java [hunk 1]
```diff
@@ -10,6 +11,7 @@
 
 import com.ichi2.anki.AnkiDroidApp;
 import com.ichi2.anki.R;
+import com.ichi2.anki.tests.InstrumentedTest;
```
[the remark this answers is not in the document]
Comment (david-allison-1) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ACRATest.java]: Interesting, no clue how that got in there! Good catch

**Comment by mikehardy** [comment]
> If you'd be fine with exchanging `lintPlayRelease` with `lint` (debug), then it'll work. `lintRelease` doesn't check `androidTest`.

That would be fine by me - I think before we had different debug vs release setups, as you mentioned, and that was the reason it wasn't just "lint" before. And of course "tested + works" is the main thing. Assuming lint target is tested+works for all java code, perfect

**Comment by mikehardy** [comment]
Just blew up your PR by merging the copyright one :)

**Comment by david-allison-1** [comment]
Okay, testing with lint:

* API ✔️ 
* API Tests ✔️ 

<details> 

```
C:\GitHub\Anki-Android-Git\Anki-Android>gradlew lint

> Task :api:lintReportDebug
Wrote HTML report to file:///C:/GitHub/Anki-Android-Git/Anki-Android/api/build/reports/lint-results-debug.html

> Task :api:lintDebug FAILED
C:\GitHub\Anki-Android-Git\Anki-Android\api\src\test\java\com\ichi2\anki\api\ApiUtilsTest.java:1: Error: All files in AnkiDroid must contain a GPLv3-compatible copyright header [MissingCopyrightHead
er from com.ichi2.anki.lint]
package com.ichi2.anki.api;
~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\GitHub\Anki-Android-Git\Anki-Android\api\src\main\java\com\ichi2\anki\FlashCardsContract.java:1: Error: All files in AnkiDroid must contain a GPLv3-compatible copyright header [MissingCopyrightHe
ader from com.ichi2.anki.lint]
package com.ichi2.anki;
~~~~~~~~~~~~~~~~~~~~~~~
2 errors, 0 warnings


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':api:lintDebug'.
> Lint found errors in the project; aborting build.

  Fix the issues identified by lint, or create a baseline to see only new errors:
  ```
  android {
      lint {
          baseline = file("lint-baseline.xml")
      }
  }
  ```

  For more details, see https://developer.android.com/studio/write/lint#snapshot

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.3.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1m 22s
56 actionable tasks: 14 executed, 42 up-to-date

C:\GitHub\Anki-Android-Git\Anki-Android>
```
</details>

Note: if API tests fail, further tests are not run. This is an edge case I'm happy with

* main
* Test
* AndroidTest

<details>

```
C:\GitHub\Anki-Android-Git\Anki-Android>gradlew lint

> Task :AnkiDroid:lintAnalyzeAmazonDebug
w: Runtime JAR files in the classpath should have the same version. These files were found in the classpath:
    C:/Users/David/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk8/1.5.30/5fd47535cc85f9e24996f939c2de6583991481b0/ko
tlin-stdlib-jdk8-1.5.30.jar (version 1.5)
    C:/Users/David/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk7/1.6.10/e1c380673654a089c4f0c9f83d0ddfdc1efdb498/ko
tlin-stdlib-jdk7-1.6.10.jar (version 1.6)
    C:/Users/David/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.6.10/b8af3fe6f1ca88526914929add63cf5e7c5049af/kotlin-
stdlib-1.6.10.jar (version 1.6)
    C:/Users/David/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.6.10/c118700e3a33c8a0d9adc920e9dec0831171925/k
otlin-stdlib-common-1.6.10.jar (version 1.6)
    C:/Users/David/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-reflect/1.6.10/1cbe9c92c12a94eea200d23c2bbaedaf3daf5132/kotlin
-reflect-1.6.10.jar (version 1.6)
w: Some runtime JAR files in the classpath have an incompatible version. Consider removing them from the classpath

> Task :AnkiDroid:lintReportAmazonDebug
w: Runtime JAR files in the classpath should have the same version. These files were found in the classpath:
    C:/Users/David/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk8/1.5.30/5fd47535cc85f9e24996f939c2de6583991481b0/ko
tlin-stdlib-jdk8-1.5.30.jar (version 1.5)
    C:/Users/David/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk7/1.6.10/e1c380673654a089c4f0c9f83d0ddfdc1efdb498/ko
tlin-stdlib-jdk7-1.6.10.jar (version 1.6)
    C:/Users/David/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.6.10/b8af3fe6f1ca88526914929add63cf5e7c5049af/kotlin-
stdlib-1.6.10.jar (version 1.6)
    C:/Users/David/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.6.10/c118700e3a33c8a0d9adc920e9dec0831171925/k
otlin-stdlib-common-1.6.10.jar (version 1.6)
    C:/Users/David/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-reflect/1.6.10/1cbe9c92c12a94eea200d23c2bbaedaf3daf5132/kotlin
-reflect-1.6.10.jar (version 1.6)
w: Some runtime JAR files in the classpath have an incompatible version. Consider removing them from the classpath
Wrote HTML report to file:///C:/GitHub/Anki-Android-Git/Anki-Android/AnkiDroid/build/reports/lint-results-amazonDebug.html

> Task :AnkiDroid:lintAmazonDebug FAILED
C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\androidTest\java\com\ichi2\anki\tests\ContentProviderTest.java:1: Error: All files in Anki
Droid must contain a GPLv3-compatible copyright header [MissingCopyrightHeader from com.ichi2.anki.lint]
package com.ichi2.anki.tests;
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\test\java\com\ichi2\testutils\MutableTime.java:1: Error: All files in AnkiDroid must conta
in a GPLv3-compatible copyright header [MissingCopyrightHeader from com.ichi2.anki.lint]
package com.ichi2.testutils;
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\main\java\com\ichi2\libanki\SortOrder.kt:1: Error: All files in AnkiDroid must contain a G
PLv3-compatible copyright header [MissingCopyrightHeader from com.ichi2.anki.lint]
package com.ichi2.libanki
~~~~~~~~~~~~~~~~~~~~~~~~~
3 errors, 0 warnings


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':AnkiDroid:lintAmazonDebug'.
> Lint found errors in the project; aborting build.

  Fix the issues identified by lint, or create a baseline to see only new errors:
  ```
  android {
      lint {
          baseline = file("lint-baseline.xml")
      }
  }
  ```

  For more details, see https://developer.android.com/studio/write/lint#snapshot

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.3.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1m 8s
61 actionable tasks: 15 executed, 1 from cache, 45 up-to-date

C:\GitHub\Anki-Android-Git\Anki-Android>
```

</details>

**Comment by david-allison-1** [comment]
Needs a re-review: I don't feel caching will work perfectly any more, but I don't feel it's viable to mix debug and release artifacts into the cache:

https://github.com/ankidroid/Anki-Android/blob/1bd7f679b965f0d40c44933db8c4631105243142/.github/workflows/lint.yml#L57-L70

I think it would be sensible to add `-debug-` to the cache string, but deferring to your expertise.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ on:
 
 jobs:
   lintRelease:
```
Comment (mikehardy) [inline .github/workflows/lint.yml:32]: nit: job title now wrong

**Comment by mikehardy** [comment]
> I don't feel caching will work perfectly any more,

Not sure why? It's a gradle cache and gradle builds into a directory tree that contains path components specifying full build type (flavor and everything). Seems like it should be fine?

**Review by mikehardy — APPROVED** [review]
All seems great to me, nice run down of the deeper issue here from the linked issue

**Comment by david-allison-1** [comment]
Ugg, one more round: `lint` is `amazon`, rather than `play`, which is likely why it's taking so long

**Comment by mikehardy** [comment]
Hmm - seems like they should take roughly equivalent times. Amazon flavor is actually a subset of Play flavor functionality so Play should be preferred as a flavor dimension in general tho

**Comment by david-allison-1** [comment]
> Hmm - seems like they should take roughly equivalent times. Amazon flavor is actually a subset of Play flavor functionality so Play should be preferred as a flavor dimension in general tho

Cached build currently performs `play`, so it won't be a warm build.

Turns out `lintPlayDebug` doesn't run `api`, so had a lot more testing. Should all be good now. Awaiting CI, then re-review

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -67,10 +67,16 @@ jobs:
...
-      - name: Run Lint Release
-        run: ./gradlew :api:lintRelease :AnkiDroid:lintPlayRelease
+      - name: Run Lint Debug
+        # "lint" is run under the 'Amazon' flavor, so slow in CI
+        # "lintPlayRelease" doesn't test androidTest
+        # "lintPlayDebug" doesn't test the API
+        run: ./gradlew lintPlayDebug
+
+      - name: Run Lint API
+        run: ./gradlew :api:lintDebug
```
Comment (mikehardy) [inline .github/workflows/lint.yml:79]: combine the two and you save all the process fork/spwan, some gradle model construction, etc

`./gradlew lintPlayDebug :api:lintDebug`

Comment (david-allison-1) [reply .github/workflows/lint.yml:79]: Any reason we don't combine all 3?

Comment (mikehardy) [reply .github/workflows/lint.yml]: yes - one is just cache warming which should be more correctly thought of as "downloads all network dependencies in a retry loop because relying on external things is CI-flaky", while the other two are "things that generate clean pass/fail signals and if they fail don't retry"

Comment (david-allison-1) [reply .github/workflows/lint.yml]: I was referring to the third check: 

```
      - name: Ktlint Check
        run: ./gradlew ktlintCheck
```

Comment (mikehardy) [reply .github/workflows/lint.yml]: :facepalm: - that passes the test of "generates clean signal, should not be retried", so yes, that one could be combined and should also provide slight performance boost
